playwright-e2e/pages/family-history.ts

```diff
@@ -0,0 +1,141 @@
+  }
+
+  async clickAddChild(): Promise<void> {
+    await this.page.getByRole('button', { name: 'Add a Child' }).click();
+  }
+  
+
+  async addFamilyMember(relationship:string, p: FamilyMember): Promise<void> {
+
+    if (relationship == 'PARENT1') {
+        await this.page.locator('mat-card-content').filter({ hasText:'Biological / Birth Parent 1' }).getByRole('button', { name: 'Edit' }).click();
+    } else if (relationship == 'PARENT2') {
+        await this.page.locator('mat-card-content').filter({ hasText:'Biological / Birth Parent 2' }).getByRole('button', { name: 'Edit' }).click();    
+    }
+    const selectorBase = 'FH_' + relationship;
+
+    await this.page.getByTestId('answer:' + selectorBase + '_ALIAS').fill(p.nickname);
+    await this.page.locator('.picklist-answer-'+ selectorBase +'_LIVING').getByText(booleanToYesOrNo(p.currentlyLiving), {exact: true}).click();
+    await this.page.locator('.picklist-answer-'+ selectorBase +'_AGE_RANGE').getByRole('combobox').selectOption(p.ageRange);
+    await this.page.locator('.picklist-answer-'+ selectorBase +'_HAD_CANCER').getByText(booleanToYesOrNo(p.cancers.length > 0), { exact:true}).click();
+
+    for (var i = 0; i < p.cancers.length; i++) {
+        if (i > 0) {
+            await this.page.getByText(' + Add another cancer').click();
+        }
+        const selectedCancer = p.cancers[i];
+        const cancerSelector = new CancerSelector(this.page,'.picklist-answer-'+ selectorBase +'_CANCER_NAME', '.picklist-answer-'+ selectorBase +'_CANCER_AGE_RANGE');
+        await cancerSelector.chooseCancer(i,selectedCancer.cancerSearch,selectedCancer.numTimesToHitDownArrow,selectedCancer.expectedCancerResult);       
+        await cancerSelector.chooseTime(i,selectedCancer.time)
+    }
+
+    if (p.ancestry.length > 0) {
+        await this.page.locator('.picklist-answer-'+ selectorBase +'_JEWISH_ANC').getByText('Yes').click();
+        for (var i = 0; i < p.ancestry.length; i++) {
+            await this.page.locator('.picklist-answer-' + selectorBase + '_JEWISH_GROUP').getByText(p.ancestry[i]).click();
+        }
+    }
+
+    var selectBoxes = 1;
+    if (p.sideOfFamily != null) {
+        let sideOfFamilyClass = 'FAMILY_SIDE_Q_3';
+        if (selectorBase.indexOf('GRAND') > 0) {
+            sideOfFamilyClass = 'FAMILY_SIDE_Q_1';
+        } else if (selectorBase.indexOf('HALF_SIBLING') > 0) {
+          sideOfFamilyClass = 'FAMILY_SIDE_Q_2';
+        }
+        await this.page.locator('.picklist-answer-' + sideOfFamilyClass).getByRole('combobox').selectOption(p.sideOfFamily)
+    }
+    if (p.sexAtBirth != null) { 
+        await this.page.locator('.picklist-answer-' + selectorBase + '_SEX_AT_BIRTH').getByRole('combobox').selectOption(p.sexAtBirth);
+    }
+    await this.page.getByRole('button', { name: 'Save' }).click();
+    await this.page.waitForResponse((resp) => resp.url().includes('/summary') && resp.status() === 200);
```

Comment:
if I'm following right, this looks like a nice way to handle a waiting for (and confirming that) a portion of a test finished without error

Reply:
Avoiding "hard timeouts" seems like it can be a challenge.  Just waiting for page elements doesn't always do the trick, so sometimes I think we have to dip into the network tab and look for other "clues" so that our waits can be smarter than just a simple "wait for a long time".